integration_tests.py

```diff
@@ -705,18 +706,38 @@ def _get_iam_token(issuer, user, password, client_id, client_secret):
     :param str client_id: client id
     :param str client_secret: client secret
     """
+    # We use subprocess instead of requests to interact with IAM
+    # Otherwise, if executed from a docker container in a different network namespace, it would not work
     query = os.path.join(issuer, "token")
-    params = {"username": user, "password": password, "grant_type": "password"}
-    response = requests.post(
+    cmd = [
+        "docker",
+        "exec",
+        "server",
+        "curl",
+        "-L",
+        "-s",
+        "-u",
+        f"{client_id}:{client_secret}",
+        "-d",
+        "grant_type=password",
+        "-d",
+        f"username={user}",
+        "-d",
+        f"password={password}",
         query,
```

Comment:
As this is repeated a lot, what do you think about abstracting this with something like:

```python
def iam_curl(url, *, data: list[str] = [], verb: Optional[str] = None, user = Optional[str], headers: list[str] = []):
    cmd = ["docker", "exec", "server", "curl", "-L", "-s"]
    if verb:
        cmd += ["-X", verb]
    if user:
        cmd += ["-u", user]
    for arg in data:
        cmd += ["-d", data]
    for header in headers:
        cmd += ["-H", header]
    cmd += [url]

    return subprocess.run(cmd, capture_output=True, check=False)
```

and then use it with something like:

```python
    ret = iam_curl(
        url,
        data=["grant_type=password", f"username={user}", f"password={password}"],
        user=f"{client_id}:{client_secret}",
    )
```

Reply:
Indeed, this is much cleaner, thanks!